src/app/security/maturity-model/components/TableOfContents.tsx

```diff
@@ -0,0 +1,46 @@
+'use client'
+
+import { clsx } from 'clsx'
+
+import { coreFunctionsData } from '../data/coreFunctionsData'
+import { scrollToSection } from '../utils/scrollToSection'
+import { useUrlHash, HASH_SIGN, type SectionHash } from '../utils/useUrlHash'
+
+export function TableOfContents() {
+  const { currentHash } = useUrlHash()
+
+  return (
+    <table role="navigation" aria-label="Table of Contents">
+      <p className="mb-4 ml-6 text-sm font-semibold uppercase text-brand-200 lg:mb-6">
+        Table of Contents
+      </p>
+
+      <ol>
+        {coreFunctionsData.map(({ slug, title }) => {
+          const sectionHash: SectionHash = `${HASH_SIGN}${slug}`
+          const isCurrentSection = currentHash === sectionHash
+
+          return (
+            <li key={slug}>
```

Comment:
Think the hover effect is missing https://www.figma.com/design/0pBfQs5tKI6bx5ypPIfVRA/FIL.org-V4-Prototypes?node-id=5961-74272&t=4CbuL2Y0K1wJdP3C-4

Reply:
Nice thanks!